### PR TITLE
[stripe] Add missing properties on StripeError

### PR DESCRIPTION
**Pull request by bruun**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/stripe/stripe-node/pull/699
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

This reflect the following change made in Stripe 7.10.0: https://github.com/stripe/stripe-node/pull/699
I also removed a property that was most likely added by mistake (`readonly: number`), and added an older missing property.

### CODE REVIEW

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -13900,9 +13900,16 @@ declare namespace Stripe {
             };
             readonly requestId: string;
             readonly detail?: any;
-            readonly: number;
             readonly params?: string;
             readonly type: string;
+            readonly statusCode?: number;
+
+            readonly charge?: charges.ICharge;
```
Comment (pushplay) [inline types/stripe/index.d.ts]: When I see this in the response JSON it's a string Charge ID.  Is this dependent upon some expand parameter?  Should it be `string | charges.ICharge`?

Does this actually get put on the IStripeError?  In my testing I see it in `raw` instead.

```javascript
stripe.charges.create(
  {amount: 1000, currency: "usd", source: "tok_chargeDeclinedFraudulent"},
  function(err, charges) {
    console.error(JSON.stringify(err, null, 2));
    console.log(charges);
  }
);

{
  "type": "StripeCardError",
  "stack": "<snip>",
  "rawType": "card_error",
  "code": "card_declined",
  "message": "Your card was declined.",
  "raw": {
    "charge": "ch_1FRllIBCvBiGc7Sd5lEOpjTG",
    "code": "card_declined",
    "decline_code": "fraudulent",
    "doc_url": "https://stripe.com/docs/error-codes/card-declined",
    "message": "Your card was declined.",
    "type": "card_error",
    "headers": {
      "server": "nginx",
      "date": "Wed, 09 Oct 2019 20:27:56 GMT",
      "content-type": "application/json",
      "content-length": "264",
      "connection": "keep-alive",
      "access-control-allow-credentials": "true",
      "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
      "access-control-allow-origin": "*",
      "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
      "access-control-max-age": "300",
      "cache-control": "no-cache, no-store",
      "request-id": "req_ocVR3raJUZZPZA",
      "stripe-version": "2019-08-14",
      "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
    },
    "statusCode": 402,
    "requestId": "req_ocVR3raJUZZPZA"
  },
  "headers": {
    "server": "nginx",
    "date": "Wed, 09 Oct 2019 20:27:56 GMT",
    "content-type": "application/json",
    "content-length": "264",
    "connection": "keep-alive",
    "access-control-allow-credentials": "true",
    "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
    "access-control-allow-origin": "*",
    "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
    "access-control-max-age": "300",
    "cache-control": "no-cache, no-store",
    "request-id": "req_ocVR3raJUZZPZA",
    "stripe-version": "2019-08-14",
    "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
  },
  "requestId": "req_ocVR3raJUZZPZA",
  "statusCode": 402
}
```

**Review by pushplay — COMMENTED** [review]
Nice.  I've been meaning to look at this one.

File: types/stripe/index.d.ts (hunk 1 above)
Comment (bruun) [reply types/stripe/index.d.ts]: Good catch @pushplay. I've only tested this with payment intents where we do get the whole Payment Intent object, so I made the assumption that the same applies to the other properties. The following docs has the correct types, I'll update the PR accordingly: https://stripe.com/docs/api/errors

Comment (bruun) [reply types/stripe/index.d.ts]: Regarding if the property is actually put on the StripeError, I've based the change on this commit on version 7.10.0 of the SDK (release recently): https://github.com/stripe/stripe-node/pull/699/commits/04c514ffef079edbb345e4e7985ba074c6783a93#diff-872c0c131ce19fb5a074d972c85c835dR40

I tested the same code, and got the following response:

```
{
  "raw": {
    "charge": "ch_1FS0iqFuhr4V1leg9qY4ygvN",
    "code": "card_declined",
    "decline_code": "fraudulent",
    "doc_url": "https://stripe.com/docs/error-codes/card-declined",
    "message": "Your card was declined.",
    "type": "card_error",
    "headers": {
      "server": "nginx",
      "date": "Thu, 10 Oct 2019 12:26:25 GMT",
      "content-type": "application/json",
      "content-length": "264",
      "connection": "keep-alive",
      "access-control-allow-credentials": "true",
      "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
      "access-control-allow-origin": "*",
      "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
      "access-control-max-age": "300",
      "cache-control": "no-cache, no-store",
      "idempotency-key": "stripe-node-retry-326def3b-ce89-457e-a138-380a8ab25d92",
      "original-request": "req_MMKmgvh6srfYAy",
      "request-id": "req_MMKmgvh6srfYAy",
      "stripe-should-retry": "false",
      "stripe-version": "2019-05-16",
      "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
    },
    "statusCode": 402,
    "requestId": "req_MMKmgvh6srfYAy"
  },
  "rawType": "card_error",
  "code": "card_declined",
  "headers": {
    "server": "nginx",
    "date": "Thu, 10 Oct 2019 12:26:25 GMT",
    "content-type": "application/json",
    "content-length": "264",
    "connection": "keep-alive",
    "access-control-allow-credentials": "true",
    "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
    "access-control-allow-origin": "*",
    "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
    "access-control-max-age": "300",
    "cache-control": "no-cache, no-store",
    "idempotency-key": "stripe-node-retry-326def3b-ce89-457e-a138-380a8ab25d92",
    "original-request": "req_MMKmgvh6srfYAy",
    "request-id": "req_MMKmgvh6srfYAy",
    "stripe-should-retry": "false",
    "stripe-version": "2019-05-16",
    "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
  },
  "requestId": "req_MMKmgvh6srfYAy",
  "statusCode": 402,
  "charge": "ch_1FS0iqFuhr4V1leg9qY4ygvN",
  "decline_code": "fraudulent"
}
```

Comment (pushplay) [reply types/stripe/index.d.ts]: Oh that charge bit is brand new then.  I'm on 7.9.0.

I think we should update the version in the header then.  I'm pretty sure we've already been incorporating changes from v7.

Comment (bruun) [reply types/stripe/index.d.ts]: Sure, I can do that as part of this PR if you want?

Comment (pushplay) [reply types/stripe/index.d.ts]: Yes please.

Comment (bruun) [reply types/stripe/index.d.ts]: Done 👍

**Pull request merged**